### ISSUE TITLE
Feature Suggestion: Standalone NPC Packs

### ISSUE THREAD

**Issue by scorpiosixnine** [focustense/easymod#143]
TL;DR: support a format which would allow mod authors to supply an NPC appearance directly to EasyNPC, instead of you having to extract it from a mod.

---

The slightlu longer version:

Right now you're jumping through a bunch of hoops to extract the relevant resources out of a bunch of mods, allow the user to pick between potential conflicts, then build everything back into a single uber-mod. 

Which is cool. 

However, you're still trying to work around a bit of a broken process. It seems to me that if you've got the technology to do what you're doing, then you could potentially build the final uber-mod from other sources than mods.

This opens up the possibility of making EasyNPC the way that people publish character overrides; by supplying "character packs" to it. These could still basically be mods under the hood, but they could sit outside of the normal Data/ root, and hence be ignored by the game.

There are a bunch of advantages:

- the packs could sit in a sub-folder of Skyrim/Data/, so they could still be installed by a mod-manager
- as they won't be at the root of Skyrim/Data/, the source packs would be ignored by the game
- each pack could contain just one NPC, will all the required resources, since EasyNPC will handle resolving conflicts, which would make it a bit easier to download just the things you want in the first place
- (though pack-bundles could also be supported for ease of distribution, or for people who really do want to download _all the things_)
- there would be no fiddly need to disable things once EasyNPC's esp has been installed, since the things packed into it won't exist in another loaded esp
- mods that need code support or other non-visual things would be free to package up a normal esp along with the "character pack" esps; they'd just both bits in the relevant places and EasyNPC would use the "character pack" and not touch the other stuff.

Naively, this feels to me like it wouldn't be an enourmous amount of work, in the great scheme of things. The "character pack" format could still basically be a mod, but maybe with a little json manifest file to add some useful metadata, and perhaps with previews bundled.

**Comment by focustense** [comment]
Interesting idea, and I agree on the "not enormous" part, although not trivial either. It requires tooling for both the end users and the mod authors.

The obvious caveats that come to my mind are:

1. Adoption. The feature's value is entirely determined by how many mod authors are willing to go along for the ride. I have in fact seen a _few_ of them in Discord who are interested in the tool or at least know it exists, but when it comes to the really big and popular mods (Bijin, Pandorable, Northbourne, etc.) it could be hit or miss. That segues into...

2. Interaction between "classic" and "special" mods. Even if new mods come out this way, the basic requirement of having to confirm the default plugin (which _does_ need to be loaded by the game) and a specific appearance doesn't go away. That is to say, the feature doesn't impact usability of the app itself, just maybe a little less work afterward on disabling the original mods and/or running post-build. But then again, maybe _more_ work if users have a harder time figuring out which mods to disable and which ones can be left alone.

3. Trying to go pack-exclusive, even in the very distant future and assuming 100% adoption, wouldn't be able to work very well with features like global replacers (e.g. #6). So no matter what, the old system doesn't really go away.

The good things are:

4. It potentially makes the modding process less error-prone, assuming some very high level of uptake in the distant future.

5. It solves the "multiple options but only one mugshot" problem. This is something you don't mention but is actually the most compelling reason to implement the feature. Authors who want to provide different options for hair, marks/tats, outfits, etc., could provide these in a FOMOD-like structure that makes it possible to provide more than one choice for the _same NPC_ in the _same mod_. That is something that's straight-up not supported today and pretty much impossible to support under the current design.

I'm not sure when or if I'd get around to this idea but it's worth keeping in the queue for that last reason especially.

**Comment by scorpiosixnine** [comment]
I guess a first pass might just be to add a second directory that it scans for "installed" mods, in addition to `Data/`? Eg a `Packs` directory alongside the `Mugshots` one. 

Any mod that was clean (eg had no dependencies) could be pulled in as possible input along with everything else, and built into the final output (if selected by the user).

As you mention above, there are a lot of mods that wouldn't be so clean, but it would give you a starting point. If any mod authors were interested, that would give them a place to deploy an alternate version of their mod to.

**Comment by scorpiosixnine** [comment]
With regard to 2 & 3, I think you could partially solve them with an additional manifest containing metadata for EasyNPC to read. 

That takes the onus off you a little to come up with clever heuristics, and puts it onto the mod author to correctly mark things up in a way that allows you to do the right thing during the build process and/or output clear instructions to the user on what mods needs to be enabled/disabled.
